include/zephyr/linker/devicetree_regions.h

```diff
@@ -290,4 +332,78 @@
  */
 #define LINKER_DT_REGION_MPU(mpu_fn) _CHECK_APPLY_FN(_DT_COMPATIBLE, _EXPAND_MPU_FN, mpu_fn)
 
+/**
+ * @brief Generate MMU regions from the device tree nodes with both
+ *        'zephyr,memory-region' and 'zephyr,memory-region-mmu' attributes.
+ *
+ * Helper macro to apply an MMU_FN macro to all the memory regions declared
+ * using the 'zephyr,memory-region-mmu' property and the 'zephyr,memory-region'
+ * compatible.
+ *
+ * @p MMU_FN must take the form:
+ *
+ * @code{.c}
+ *   #define MMU_FN(name, base, size, attr) ...
+ * @endcode
+ *
+ * The 'name', 'base' and 'size' parameters are taken from the DT node.
+ *
+ * The 'zephyr,memory-region-mmu' string array property is passed as an extended token
+ * to the MMU_FN macro using the 'attr' parameter, mmu property is defined
+ * in architecture MMU header, for example: include/zephyr/arch/arm64/arm_mmu.h
+ *
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     ccm: ccm@30380000 {
+ *                         compatible = "nxp,imx-ccm";
+ *                         reg = <0x30380000 DT_SIZE_K(64)>;
+ *                         zephyr,memory-region = "CCM";
+ *                         zephyr,memory-region-mmu = "MT_DEVICE_nGnRnE", "MT_P_RW_U_NA", "MT_NS";
+ *                     };
+ *             };
+ *     };
+ *
+ * For detailed information about MMU region attribute define configuration refer
+ * to the specific architecture MMU header.
+ * For example: include/zephyr/arch/arm64/arm_mmu.h.
+ *
+ * The 'attr' parameter of the MMU_FN function will be the or-ed value for all valid
+ * C token of properity of "zephyr,memory-region-mmu", these C token is defined
+ * in architecture MMU header, for example: include/zephyr/arch/arm64/arm_mmu.h
+ *
+ * Example:
+ *
+ * @code{.c}
+ *
+ *   #define MPU_FN(p_name, p_base, p_size, p_attr) \
+ *       {                                          \
+ *           .name = p_name,                        \
+ *           .base = p_base,                        \
+ *           .size = p_size,                        \
+ *           .attr = p_attr,                        \
+ *       }
+ *
+ *   static const struct arm_mpu_region mpu_regions[] = {
+ *       ...
+ *       LINKER_DT_REGION_MPU(MPU_FN)
+ *       ...
+ *   };
+ * @endcode
+ *
+ * With dts example, the code will be:
+ *   static const struct arm_mpu_region mpu_regions[] = {
+ *       ...
+ *       { .name = "CCM",
+ *         .base = 0x30380000,
+ *         .size = DT_SIZE_K(64),
+ *         .attr = MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS, },
+ *       ...
+ *   };
+ *
+ */
+#define LINKER_DT_REGION_MMU(mmu_fn) _CHECK_APPLY_MMU_FN(_EXPAND_MMU_FN, mmu_fn)
```

Comment:
At this point I'm realizing how misnamed are `LINKER_DT_REGION_MPU` and this macro  because these macros have nothing to do with the linker script.

Can you rename those to something like `BUILD_DT_REGION_{MPU,MMU}`?

Reply:
Done.